main.tf

```diff
@@ -1,5 +1,6 @@
 locals {
-  postfix = var.name_postfix != "" ? var.name_postfix : random_id.id.hex
+  postfix            = var.name_postfix != "" ? var.name_postfix : random_id.id.hex
+  planCredentialPort = replace(var.plan, "standalone", "") != var.plan ? "port" : "sentinel_port"
```

Comment:
Here I would check the value of "port" instead of looking at the plan value

Reply:
Clarifying a bit, the trick is that the credential object is different according to the plan, so I'm getting errors likes "This object has no argument, nested block, or exported attribute named sentinel_port" when trying to get the value of the port, so even functions like try(), coalesce(), etc the error remains.

Working using the plan it's not ideal because we need to keep track of newer plans but it worked nice.